Realm/Realm.Sync/ProgressNotifications/SyncProgressObservable.cs

```diff
@@ -73,6 +75,11 @@ public void Notify(ulong transferredBytes, ulong transferableBytes)
                 Task.Run(() =>
                 {
                     _observer.OnNext(new SyncProgress(transferredBytes, transferableBytes));
+
+                    if (_mode == ProgressMode.ForCurrentlyOutstandingWork && transferableBytes == transferredBytes)
```

Comment:
I think this should be `transferableBytes <= transferredBytes` just in case there's some overhead and we go past the target. Or, if you prefer, swap the order  `transferredBytes >= transferableBytes`.